src/cljs/main/broadfcui/main.cljs

```diff
@@ -178,29 +179,10 @@
    (fn [{:keys [this state]}]
      (init-nav-paths)
      (this :handle-hash-change)
-     (set! js/window.forceSignedIn
-           (fn [auth-token]
-             (utils/ajax {:url (str "https://www.googleapis.com/oauth2/v3/tokeninfo?access_token=" auth-token)
-                          :on-done
-                          (fn [{:keys [status-code success? get-parsed-response]}]
-                            (if success?
-                              (let [{:keys [email sub]} (get-parsed-response)
-                                    instance (clj->js
-                                              {:currentUser
-                                               {:get
-                                                (fn []
-                                                  (clj->js {:getAuthResponse
-                                                            (fn [] (clj->js {:access_token auth-token}))
-                                                            :getBasicProfile
-                                                            (fn [] (clj->js
-                                                                    {:getEmail (fn [] email)
-                                                                     :getId (fn [] sub)}))}))
-                                                :listen (fn [])}
-                                               :signOut
-                                               (fn [] (swap! state update :user-status disj :signed-in))})]
-                                (utils/set-google-auth2-instance! instance)
-                                (swap! state update :user-status conj :signed-in))
-                              (utils/log "Failed to validate token: " status-code)))}))))
+     (set! (.-forceSignedIn js/window)
+           (auth/force-signed-in {:on-sign-in #(swap! state update :user-status conj :signed-in)
+                                  :on-sign-out #(swap! state update :user-status disj :signed-in)
+                                  :on-error (fn [error] (swap! state assoc :force-sign-in-error error))})))
```

Comment:
`#(swap! state assoc :force-sign-in-error %)` would be more concise, but possibly less clear. Your call.

Reply:
Maybe only less clear to learners like me. I've been bothered by the style being inconsistent with `:on-sign-in` and `:on-sign-out` so I like this change.